solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,356 @@
+  default     = "private"
+  validation {
+    condition     = can(regex("public|private", var.key_management_service_endpoint_type))
+    error_message = "The specified KMS endpoint type is not supported. The following values are supported: `public` or `private`."
+  }
+}
+
+variable "event_notifications_key_name" {
+  type        = string
+  default     = "en-key"
+  description = "The name for the key that will be created for the Event Notifications instance. Not used if an existing key is specified. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+}
+
+variable "event_notifications_key_ring_name" {
+  type        = string
+  default     = "en-key-ring"
+  description = "The name of the key ring which will be created for the Event Notifications instance. Not used if supplying an existing key. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+}
+
+variable "cloud_object_storage_key_ring_name" {
+  type        = string
+  default     = "en-cos-key-ring"
+  description = "The name of the key ring which will be created for Object Storage. Not used if supplying an existing key or if `existing_cloud_object_storage_bucket_name` is specified. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+}
+
+variable "cloud_object_storage_key_name" {
+  type        = string
+  default     = "en-cos-key"
+  description = "The name of the key which will be created for the Event Notifications. Not used if supplying an existing key. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+}
+
+variable "skip_event_notifications_key_management_service_auth_policy" {
+  type        = bool
+  description = "Set to true to skip the creation of an IAM authorization policy that permits the Event Notifications instance to read the encryption key from the KMS instance. If a value is specified for `ibmcloud_key_management_service_api_key`, the policy is created in the KMS account."
+  default     = false
+}
+
+variable "ibmcloud_key_management_service_api_key" {
+  type        = string
+  description = "The IBM Cloud API key that can create a root key and key ring in the key management service (KMS) instance. If not specified, the 'ibmcloud_api_key' variable is used. Specify this key if the instance in `existing_key_management_service_instance_crn` is in an account that's different from the Event Notifications instance. Leave this input empty if the same account owns both instances."
+  sensitive   = true
+  default     = null
+}
+
+########################################################################################################################
+# COS
+########################################################################################################################
+
+variable "cloud_object_storage_integration_enabled" {
```

Comment:
Can we rename this to `enable_collecting_failed_events` and ensure to add variable validation to ensure COS details are added if set to true. Also update the variable description to mention what inputs are required if setting to true.